### ISSUE TITLE
error while loading shared libraries: libmpfr.so

### ISSUE THREAD

**Issue by cansi** [ValveSoftware/steamlink-sdk#22]
I'm clearly doing something wrong. I'm on Debian 10 Buster trying to compile /examples/linux/example.c, easy thing I thought. Opened bash, sourced setenv.sh, then tried to invoke the compiler like this:

`> $CC -c example.c`

But the resulting output was:

`/home/user/steamlink-sdk/toolchain/bin/../libexec/gcc/armv7a-cros-linux-gnueabi/9.2.0/cc1: error while loading shared libraries: libmpfr.so.4: cannot open shared object file: No such file or directory`

I can't find any trace of that lib in the SDK, any ideas of what I'm missing here? Thank you in advance for your time.

**Comment by yuri-becker** [comment]
I was just facing the same issue trying to compile on Ubuntu 18 running in the Windows Subsystem for Linux.

I'll assume you have libmpfr installed. If not, the package is called libmpfr6 or libmpfr-dev in the Ubuntu repositories, I'll assume its the same on Debian. You can simply install it via `sudo apt install libmpfr-dev`.

The error still appeared for me though - although when running `ldconfig -v` libmpfr.so.6 showed up.

I tried various things, but [this workaround](https://stackoverflow.com/a/51657518) ended up doing it for me. I don't really understand why that is needed, but maybe someone can explain to me?

**tl;dr** [https://stackoverflow.com/a/51657518](https://stackoverflow.com/a/51657518)

**Comment by slouken** [comment]
I just added the dependencies in the toolchain/lib directory. Does that fix it automatically? If not, you should be able to set your LD_LIBRARY_PATH to that directory and have it work.

**Comment by cansi** [comment]
I had already installed libmpfr-dev on my system, but I wasn't sure if that would be the right thing to do. Adding the symlink did the trick anyway, not sure if setting LD_LIBRARY_PATH would have make the same effect. I'll close the issue, thank you all!
